api-put-object-multipart.go

```diff
@@ -190,6 +191,10 @@ func (c Client) putObjectMultipartStream(bucketName, objectName string, reader i
 	for i := 1; i <= totalPartsCount; i++ {
```

Comment:
this will also cause data corruption if previous incomplete upload had more parts than the correct upload. you can use:
`for i := 1; i <= partNumber; i++ {` to avoid that situation and also the `if size < 0 {` check below will not be needed.